compose.c

```diff
@@ -862,6 +902,42 @@ static int draw_envelope_addr(int field, struct AddressList *al,
   return lines_used;
 }
 
+static int draw_envelope_user_hdrs(const struct ComposeRedrawData *rd, int row)
+{
+  const char *overflow_text = "...";
+  int rows_used = 0;
+
+  struct ListNode *first = STAILQ_FIRST(&rd->email->env->userhdrs);
+  if (first)
+  {
+    /* Draw first entry on same line as prompt */
+    draw_header(rd->win_envelope, row, HDR_CUSTOM_HEADERS);
+    mutt_paddstr(rd->win_envelope->state.cols -
```

Comment:
I feel like there's definitely a simpler solution here. With the line above printing:
```
        Headers: 
```
I'm just wanting to add `first->data` directly after this, but truncated if wider than the window width

Reply:
That looks right.
`draw_header()` manages the colour and alignment, so that needs to be separate.